src/ast.rs

```diff
@@ -1236,83 +1102,18 @@ impl AstStatement {
-                true if is_covered_by!(i16, *value) => Some(INT_TYPE),
-                true if is_covered_by!(i32, *value) => Some(DINT_TYPE),
-                true if is_covered_by!(i64, *value) => Some(LINT_TYPE),
-
-                false if is_covered_by!(u8, *value) => Some(USINT_TYPE),
-                false if is_covered_by!(u16, *value) => Some(UINT_TYPE),
-                false if is_covered_by!(u32, *value) => Some(UDINT_TYPE),
-                false if is_covered_by!(u64, *value) => Some(ULINT_TYPE),
-                _ => Some(VOID_TYPE),
-            },
-            AstStatement::LiteralBool { .. } => Some(BOOL_TYPE),
-            AstStatement::LiteralString { is_wide: true, .. } => Some(WSTRING_TYPE),
-            AstStatement::LiteralString { is_wide: false, .. } => Some(STRING_TYPE),
-            AstStatement::LiteralReal { .. } => Some(LREAL_TYPE),
-            AstStatement::LiteralDate { .. } => Some(DATE_TYPE),
-            AstStatement::LiteralDateAndTime { .. } => Some(DATE_AND_TIME_TYPE),
-            AstStatement::LiteralTime { .. } => Some(TIME_TYPE),
-            AstStatement::LiteralTimeOfDay { .. } => Some(TIME_OF_DAY_TYPE),
-            _ => None,
-        }
-    }
-
     /// returns true if this AST Statement is a literal or reference that can be
     /// prefixed with a type-cast (e.g. INT#23)
     pub fn is_cast_prefix_eligible(&self) -> bool {
-        matches!(
-            self,
-            AstStatement::LiteralBool { .. }
-                | AstStatement::LiteralInteger { .. }
-                | AstStatement::LiteralReal { .. }
-                | AstStatement::LiteralString { .. }
-                | AstStatement::LiteralTime { .. }
-                | AstStatement::LiteralDate { .. }
-                | AstStatement::LiteralTimeOfDay { .. }
-                | AstStatement::LiteralDateAndTime { .. }
-                | AstStatement::Reference { .. }
-        )
-    }
-    /// Returns true if the current statement is a literal
-    pub fn is_literal(&self) -> bool {
-        matches!(
-            self,
-            AstStatement::LiteralNull { .. }
-                | AstStatement::LiteralInteger { .. }
-                | AstStatement::LiteralDate { .. }
-                | AstStatement::LiteralDateAndTime { .. }
-                | AstStatement::LiteralTimeOfDay { .. }
-                | AstStatement::LiteralTime { .. }
-                | AstStatement::LiteralReal { .. }
-                | AstStatement::LiteralBool { .. }
-                | AstStatement::LiteralString { .. }
-                | AstStatement::LiteralArray { .. }
-        )
+        // TODO: figure out a better name for this...
+        match self {
+            AstStatement::Literal { kind, .. } => kind.is_cast_prefix_eligible(),
+            AstStatement::Reference { .. } => true,
+            _ => false,
+        }
     }
 
+    /// Returns true if the current statement is a literal
```

Comment:
This comment seems to be in the wrong place/obsolete. Did you mean to delete it along with the `is_literal` function?